File: iosApp/iosApp/Analytics/AnalyticsProvider.swift
//
//  AnalyticsProvider.swift
//  iosApp
//
//  Created by Brandon Rodriguez on 6/12/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import FirebaseAnalytics
import Foundation
import Shared

class AnalyticsProvider: Shared.Analytics {
    static let shared = AnalyticsProvider()

    override func logEvent(name: String, parameters: [String: String]) {
        FirebaseAnalytics.Analytics.logEvent(name, parameters: parameters)
    }

    override func setUserProperty(name: String, value: String) {
        FirebaseAnalytics.Analytics.setUserProperty(value, forName: name)
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/analytics/AnalyticsProvider.kt
package com.mbta.tid.mbta_app.android.analytics

import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.logEvent
import com.mbta.tid.mbta_app.analytics.Analytics

class AnalyticsProvider(private val firebaseAnalytics: FirebaseAnalytics) : Analytics() {
    override fun logEvent(name: String, parameters: Map<String, String>) {
        firebaseAnalytics.logEvent(name) {
            for ((key, value) in parameters) {
                param(key, value)
            }
        }
    }

    override fun setUserProperty(name: String, value: String) {
        firebaseAnalytics.setUserProperty(name, value)
    }
}
